posthog/queries/event_query/event_query.py

```diff
@@ -78,12 +77,12 @@ def __init__(
         self._extra_fields = extra_fields
         self._person_on_events_mode = person_on_events_mode
 
-        # Guards against a ClickHouse bug involving multiple joins against the same table
-        # with the same column name. This issue manifests for us with formulas, where on queries A and B we join events against itself
-        # and both tables end up having $session_id. Without a formula this is not a problem.
+        # Guards against a ClickHouse bug involving multiple joins against the same table with the same column name.
+        # This issue manifests for us with formulas, where on queries A and B we join events against itself
+        # and both tables end up having $session_id. Without a formula this is not a problem.]
         self._session_id_alias = (
-            f"session_id_{random.randint(0, 10000000000000000000)}"
-            if hasattr(self._filter, "formula") and self._filter.formula is not None # type: ignore
+            f"session_id_{self._entity.index}"
```

Comment:
lol thanks @Twixes I looked into using entity but wasn't aware it had the index and had it acessible here

Reply:
I'm allergic to non-deterministic tests, had to think of a better alternative 😄